File: DSA_TypeScript/Math/HCF/divideMethod.ts
const divideMothod = (nums:Array<number>)=>{
    let hcf =  hcfOfTwoNums(nums[0], nums[1])
    for(let i=2; i<nums.length; i++){
        hcf = hcfOfTwoNums(nums[i], hcf);
    }
    return hcf
}

const hcfOfTwoNums  = (num1:number, num2:number)=>{

    while(num1>0 && num2>0)
    {
   
        if(num1>num2){
            num1 = num1%num2
        }else{
            num2 = num2%num1
        }
    }
    return num1||num2
}


export default divideMothod;
// 
// console.log(divideMothod([84,90]))
// console.log(divideMothod([45, 20,20, 45])).

File: DSA_Python/Math/HCF/divideMethod.py

def divideMethod (nums):
    hcf  = hcfOfTwoNums(nums[0], nums[1])
    limit = 0
   
    for i in nums :
        if limit>1 :
            hcf = hcfOfTwoNums(i, hcf)
        limit+=1
    return hcf

def hcfOfTwoNums (n1,n2) :
    while n1>0 and n2>0 :
        if n1>n2 :
            n1 = n1 % n2 
        else :
            n2 = n2 % n1
    return n1 or n2      


print(divideMethod([84,90]))